diff --git a/build/serializer/serializeChunks.js b/build/serializer/serializeChunks.js
--- a/build/serializer/serializeChunks.js
+++ b/build/serializer/serializeChunks.js
@@ -563,7 +563,7 @@
     while (toCompare.length) {
         const chunk = toCompare.shift();
         for (const chunk2 of toCompare) {
-            if (chunk !== chunk2 && chunk.isAsync && chunk2.isAsync) {
+            if (chunk !== chunk2 && chunk.isAsync && chunk2.isAsync && !chunk.isEntry && !chunk2.isEntry) {
                 for (const dep of chunk.deps) {
                     if (chunk2.deps.has(dep)) {
                         chunk.deps.delete(dep);
diff --git a/build/transform-worker/collect-dependencies.js b/build/transform-worker/collect-dependencies.js
--- a/build/transform-worker/collect-dependencies.js
+++ b/build/transform-worker/collect-dependencies.js
@@ -97,9 +97,11 @@
                     // Replace `new Worker` invocation with `asyncRequire.unstable_createWorker` helper call,
                     // creating an Object URL with a shim script that remaps `importScripts` and `fetch`, while calling the entrypoint with `importScripts`.
                     // This works around COEP/CORP issues, since worker entrypoint scripts aren't necessarily expected to match these headers
-                    path.get('callee').replaceWith(makeCreateWorkerTemplate({
-                        ASYNC_REQUIRE_MODULE_PATH: nullthrows(state.asyncRequireModulePathStringLiteral),
-                    }));
+                    if (state.collectOnly !== true) {
+                        path.get('callee').replaceWith(makeCreateWorkerTemplate({
+                            ASYNC_REQUIRE_MODULE_PATH: nullthrows(state.asyncRequireModulePathStringLiteral),
+                        }));
+                    }
                 }
             }
         },
