diff --git a/src/async-require/asyncRequireModule.ts b/src/async-require/asyncRequireModule.ts
--- a/src/async-require/asyncRequireModule.ts
+++ b/src/async-require/asyncRequireModule.ts
@@ -157,7 +157,7 @@
     try {
       const content = makeWorkerContent(workerUrl);
       workerUrl = URL.createObjectURL(new Blob([content], { type: 'text/javascript' }));
-      return new Worker(workerUrl, workerOpts);
+      return new Worker(workerUrl, { ...workerOpts, type: 'classic' });
     } finally {
       URL.revokeObjectURL(workerUrl);
     }
