diff --git a/lib/commonjs/gesture-handler/ScreenGestureDetector.js b/lib/commonjs/gesture-handler/ScreenGestureDetector.js
index 0a520d6..e58ed80 100644
--- a/lib/commonjs/gesture-handler/ScreenGestureDetector.js
+++ b/lib/commonjs/gesture-handler/ScreenGestureDetector.js
@@ -180,13 +180,13 @@ const ScreenGestureDetector = ({
   }
   let panGesture = _reactNativeGestureHandler.Gesture.Pan().onStart(onStart).onUpdate(onUpdate).onEnd(onEnd);
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture.activeOffsetX(ACTIVATION_DISTANCE).hitSlop({
         left: 0,
         top: 0,
-        width: HIT_SLOP_SIZE
+        width: _reactNative.Dimensions.get('window').width * 0.8
       });
     } else if (goBackGesture === 'swipeLeft') {
       panGesture = panGesture.activeOffsetX(-ACTIVATION_DISTANCE).hitSlop({
diff --git a/lib/module/gesture-handler/ScreenGestureDetector.js b/lib/module/gesture-handler/ScreenGestureDetector.js
index b6cfc6f..fc51579 100644
--- a/lib/module/gesture-handler/ScreenGestureDetector.js
+++ b/lib/module/gesture-handler/ScreenGestureDetector.js
@@ -172,13 +172,13 @@ const ScreenGestureDetector = ({
   }
   let panGesture = Gesture.Pan().onStart(onStart).onUpdate(onUpdate).onEnd(onEnd);
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture.activeOffsetX(ACTIVATION_DISTANCE).hitSlop({
         left: 0,
         top: 0,
-        width: HIT_SLOP_SIZE
+        width: Dimensions.get('window').width * 0.8
       });
     } else if (goBackGesture === 'swipeLeft') {
       panGesture = panGesture.activeOffsetX(-ACTIVATION_DISTANCE).hitSlop({
diff --git a/src/gesture-handler/ScreenGestureDetector.tsx b/src/gesture-handler/ScreenGestureDetector.tsx
index 2afab56..d73134b 100644
--- a/src/gesture-handler/ScreenGestureDetector.tsx
+++ b/src/gesture-handler/ScreenGestureDetector.tsx
@@ -224,12 +224,12 @@ const ScreenGestureDetector = ({
     .onEnd(onEnd);
 
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture
         .activeOffsetX(ACTIVATION_DISTANCE)
-        .hitSlop({ left: 0, top: 0, width: HIT_SLOP_SIZE });
+        .hitSlop({ left: 0, top: 0, width: Dimensions.get('window').width * 0.8 });
     } else if (goBackGesture === 'swipeLeft') {
       panGesture = panGesture
         .activeOffsetX(-ACTIVATION_DISTANCE)
