diff --git a/lib/commonjs/gesture-handler/ScreenGestureDetector.js b/lib/commonjs/gesture-handler/ScreenGestureDetector.js
index 0a520d60713c64667fe2b2445ec2895a78354b08..126e17e6d2c4d9e912c2cf16f74911f29154071e 100644
--- a/lib/commonjs/gesture-handler/ScreenGestureDetector.js
+++ b/lib/commonjs/gesture-handler/ScreenGestureDetector.js
@@ -180,7 +180,7 @@ const ScreenGestureDetector = ({
   }
   let panGesture = _reactNativeGestureHandler.Gesture.Pan().onStart(onStart).onUpdate(onUpdate).onEnd(onEnd);
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture.activeOffsetX(ACTIVATION_DISTANCE).hitSlop({
diff --git a/lib/module/gesture-handler/ScreenGestureDetector.js b/lib/module/gesture-handler/ScreenGestureDetector.js
index b6cfc6fefe27ae140748fab6f5d927fe9bc603fb..3969c3ae11bc055af91a87e69f5d2e7ff2ae18b1 100644
--- a/lib/module/gesture-handler/ScreenGestureDetector.js
+++ b/lib/module/gesture-handler/ScreenGestureDetector.js
@@ -172,7 +172,7 @@ const ScreenGestureDetector = ({
   }
   let panGesture = Gesture.Pan().onStart(onStart).onUpdate(onUpdate).onEnd(onEnd);
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture.activeOffsetX(ACTIVATION_DISTANCE).hitSlop({
diff --git a/src/gesture-handler/ScreenGestureDetector.tsx b/src/gesture-handler/ScreenGestureDetector.tsx
index 2afab563098831c3d92151353058626e357b959a..283a2d560ac4259ca264e49af4378909f0a8c4d3 100644
--- a/src/gesture-handler/ScreenGestureDetector.tsx
+++ b/src/gesture-handler/ScreenGestureDetector.tsx
@@ -224,7 +224,7 @@ const ScreenGestureDetector = ({
     .onEnd(onEnd);
 
   if (screenEdgeGesture) {
-    const HIT_SLOP_SIZE = 50;
+    const HIT_SLOP_SIZE = 120;
     const ACTIVATION_DISTANCE = 30;
     if (goBackGesture === 'swipeRight') {
       panGesture = panGesture
